appformer-form-modeler/appformer-form-modeler-codegen/appformer-form-modeler-codegen-impl/src/test/java/org/kie/appformer/formmodeler/codegen/view/impl/inputs/impl/TextBoxHelperTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kie.appformer.formmodeler.codegen.view.impl.inputs.impl;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.appformer.formmodeler.codegen.view.impl.java.inputs.impl.TextBoxHelper;
+import org.kie.appformer.formmodeler.codegen.view.impl.java.inputs.util.SourceGenerationValueConvertersFactory;
+import org.kie.workbench.common.forms.fields.shared.fieldTypes.basic.textBox.definition.TextBoxBaseDefinition;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.junit.Assert.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TextBoxHelperTest {
+
+    private static final String INPUT_WIDGET_CLASS_NAME = "org.gwtbootstrap3.client.ui.TextBox";
+    TextBoxHelper helper;
+
+    @Mock
+    TextBoxBaseDefinition mockFieldDefinition;
+
+    @Before
+    public void initHelper() {
+        helper = new TextBoxHelper();
+    }
+
+    @Test
+    public void testGetSupportedFieldTypeCode() {
+        String expectedTypeCode = TextBoxBaseDefinition.FIELD_TYPE.getTypeName();
+        String actualTypeCode = helper.getSupportedFieldTypeCode();
+        assertEquals(actualTypeCode,
+                     expectedTypeCode);
+    }
+
+    @Test
+    public void testGetInputWidget() {
+        String expectedInputWidget = INPUT_WIDGET_CLASS_NAME;
+        String actualInputWidget = helper.getInputWidget(null);
+        assertEquals(actualInputWidget,
+                     expectedInputWidget);
+    }
+
+    @Test
+    public void testGetReadOnlyMethod() {
+
+        String fieldName = "testFieldName";
+        String readOnlyParam = "readOnly";
+        String actualString = helper.getReadonlyMethod(fieldName,
+                                                       readOnlyParam);
+
+        String expectedString = fieldName + ".setReadOnly( " + readOnlyParam + ");";
+
+        assertEquals(actualString,
+                     expectedString);
+    }
+
+    @Test
+    public void testGetConverterClassNameString() {
+        testGetConverterClassName(String.class);
+    }
+
+    @Test
+    public void testGetConverterClassNameBigCharacter() {
+        testGetConverterClassName(Character.class);
+    }
+
+    @Test
+    public void testGetConverterClassNameBigChar() {
```

Comment:
Based on the name of the JIRA, it seems like we should have regression tests here for Short, BigDecimal, Byte, etc.